lib/dotcom/system_status/groups.ex

```diff
@@ -0,0 +1,337 @@
+defmodule Dotcom.SystemStatus.Groups do
+  @moduledoc """
+  A module that groups alerts into statuses for the system status widget.
+  """
+
+  alias Alerts.Alert
+
+  # &groups/2 returns an ordered data structure, sorted in the order
+  # given by `@lines`.
+  @lines ["Blue", "Orange", "Red", "Green"]
+  @green_line_branches ["Green-B", "Green-C", "Green-D", "Green-E"]
+  @routes ["Blue", "Mattapan", "Orange", "Red"] ++ @green_line_branches
+
+  def groups(alerts, time) do
+    grouped_alerts = Map.new(@routes, &{&1, alerts_for_line(alerts, &1)})
```

Comment:
That function should be `alerts_for_route` if it operates per-route, no?

I also wonder if something in the `Alerts.Match` module might meet your needs, what if `Alerts.Match.match(alerts, %InformedEntity{route: &1})`?

Reply:
> That function should be `alerts_for_route` if it operates per-route, no?

Yep! Done.

> I also wonder if something in the `Alerts.Match` module might meet your needs, what if `Alerts.Match.match(alerts, %InformedEntity{route: &1})`?

Ugh - I agree, but I've been struggling to actually get `Alerts.Match` to work, and I'm not sure exactly why. I still think it's worth using the machinery that already exists, but would you mind if I did that as a follow-up instead?